x-pack/plugins/infra/common/http_api/inventory_views/v1/common.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { nonEmptyStringRt } from '@kbn/io-ts-utils';
+import * as rt from 'io-ts';
+
+export const INVENTORY_VIEW_URL = '/api/infra/inventory_views';
+export const INVENTORY_VIEW_URL_ENTITY = `${INVENTORY_VIEW_URL}/{inventoryViewId}`;
+export const getInventoryViewUrl = (inventoryViewId?: string) =>
+  [INVENTORY_VIEW_URL, inventoryViewId].filter(Boolean).join('/');
+
+export const inventoryViewRequestParamsRT = rt.type({
+  inventoryViewId: rt.string,
+});
+
+export type InventoryViewRequestParams = rt.TypeOf<typeof inventoryViewRequestParamsRT>;
+
+export const inventoryViewRequestQueryRT = rt.partial({
+  sourceId: rt.string,
+});
+
+export type InventoryViewRequestQuery = rt.TypeOf<typeof inventoryViewRequestQueryRT>;
+
+const inventoryViewAttributesResponseRT = rt.intersection([
```

Comment:
**note**: for the attributes of this saved object, the configuration has always been dynamic and not strictly typed. As they can vary depending on the user usage on the page, I left it open to all the attributes as it was before, marking only those that I'm sure will be present in the response.